crates/services/src/seqlock.rs

```diff
@@ -0,0 +1,158 @@
+//! A simple implementation of a sequential lock.
+//! More details: <https://docs.kernel.org/locking/seqlock.html>
+
+use std::{
+    cell::UnsafeCell,
+    sync::atomic::{
+        AtomicU64,
+        Ordering,
+    },
+};
+
+/// A simple implementation of a sequential lock.
+/// some usage of unsafe, T must be Copy
+#[derive(Debug)]
+pub struct SeqLock<T> {
+    sequence: AtomicU64,
+    data: UnsafeCell<T>,
+}
+
+unsafe impl<T: Send> Sync for SeqLock<T> {}
+
+/// The writer handle for the `SeqLock`.
+/// Only one writer exists for a `SeqLock`.
+#[derive(Debug)]
```

Comment:
The docs should mention that `Clone` bound is missing to prevent creating multiple writers.

Reply:
addressed in [6d626b5](https://github.com/FuelLabs/fuel-core/pull/2524/commits/6d626b582438a4923cbfa4a399a03ee22ac6ffee)